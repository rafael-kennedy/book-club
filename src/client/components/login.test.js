import { mountWithInjections } from "../../test/utils/mount-with-injections";

import subject from "./login.vue";

describe("login component", () => {
  test("displays button when not logged in", () => {
    const wrapper = mountWithInjections(subject, {
      propsData: { value: null }
    });
    expect(wrapper).toMatchSnapshot();
  });

  test("displays logout button when logged in", () => {
    const wrapper = mountWithInjections(subject, {
      propsData: { value: { email: "fake" } }
    });
    expect(wrapper).toMatchSnapshot();
  });

  test("makes login call when user logs in", () => {
    const wrapper = mountWithInjections(subject, {
      propsData: { value: null }
    });
    expect(wrapper).toMatchSnapshot();
  });

  test("makes register when user tries to register", () => {
    const wrapper = mountWithInjections(subject, {
      propsData: { value: null }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
