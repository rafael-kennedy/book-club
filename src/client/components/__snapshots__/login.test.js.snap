// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component displays button when not logged in 1`] = `
<div>
  <uiiconbutton-stub icon="account_box" dropdown-position="bottom right" type="secondary"></uiiconbutton-stub>
  <uimodal-stub title="Login">
    <div class="default-margin dropdown-reposition">
      <uitextbox-stub floating-label="" label="Email" placeholder="Enter your email address"></uitextbox-stub>
      <uitextbox-stub floating-label="" label="Password" type="password" placeholder="Enter your password"></uitextbox-stub>
      <uibutton-stub disabled="true" color="primary">
        Login
      </uibutton-stub>
      Don't have an account yet?
      <uibutton-stub disabled="true">
        Register
      </uibutton-stub>
    </div>
  </uimodal-stub>
</div>
`;

exports[`login component displays logout button when logged in 1`] = `
<div>
  <uiiconbutton-stub icon="account_box" dropdown-position="bottom right" type="secondary"></uiiconbutton-stub>
  <uimodal-stub title="Login">
    <div>
      <uibutton-stub>
        Logout
      </uibutton-stub>
    </div>
  </uimodal-stub>
</div>
`;

exports[`login component makes login call when user logs in 1`] = `
<div>
  <uiiconbutton-stub icon="account_box" dropdown-position="bottom right" type="secondary"></uiiconbutton-stub>
  <uimodal-stub title="Login">
    <div class="default-margin dropdown-reposition">
      <uitextbox-stub floating-label="" label="Email" placeholder="Enter your email address"></uitextbox-stub>
      <uitextbox-stub floating-label="" label="Password" type="password" placeholder="Enter your password"></uitextbox-stub>
      <uibutton-stub disabled="true" color="primary">
        Login
      </uibutton-stub>
      Don't have an account yet?
      <uibutton-stub disabled="true">
        Register
      </uibutton-stub>
    </div>
  </uimodal-stub>
</div>
`;

exports[`login component makes register when user tries to register 1`] = `
<div>
  <uiiconbutton-stub icon="account_box" dropdown-position="bottom right" type="secondary"></uiiconbutton-stub>
  <uimodal-stub title="Login">
    <div class="default-margin dropdown-reposition">
      <uitextbox-stub floating-label="" label="Email" placeholder="Enter your email address"></uitextbox-stub>
      <uitextbox-stub floating-label="" label="Password" type="password" placeholder="Enter your password"></uitextbox-stub>
      <uibutton-stub disabled="true" color="primary">
        Login
      </uibutton-stub>
      Don't have an account yet?
      <uibutton-stub disabled="true">
        Register
      </uibutton-stub>
    </div>
  </uimodal-stub>
</div>
`;
